src/WebJobs.Script/Description/DotNet/DotNetCompilationServiceFactory.cs

```diff
@@ -24,14 +27,31 @@ ImmutableArray<ScriptType> ICompilationServiceFactory.SupportedScriptTypes
             }
         }
 
+        private static OptimizationLevel OptimizationLevel
+        {
+            get
+            {
+                if (_optimizationLevel == null)
+                {
+                    string releaseModeSetting = Environment.GetEnvironmentVariable(DotNetConstants.CompilationReleaseMode);
+
+                    _optimizationLevel = string.Equals(releaseModeSetting, bool.TrueString, StringComparison.OrdinalIgnoreCase)
```

Comment:
Another thing I debated on :)
This aligns well with what both .NET languages we support expose and the way this is surfaced in other products as well (e.g. ASP.NET). That's why the option was `...CompilationReleaseMode` as opposed to `...CompilationMode"